### PR TITLE
SDL_JoystickId is int32

### PR DESCRIPTION
**Pull request by ftsf**
I encountered an issue where my Joystick axis wasn't registering.
According to https://hg.libsdl.org/SDL/file/e12c38730512/include/SDL_joystick.h#l72 it should be an int32 rather than uint8.
Changing this got the joystick axis to work.

### CODE REVIEW

File: src/sdl2.nim [hunk 1]
```diff
@@ -116,15 +116,16 @@ type
   JoyAxisEventObj* = object
     kind*: EventType
     timestamp*: uint32
-    which*: uint8
```
Comment (yglukhov) [inline src/sdl2.nim]: `pad3` missing. Here's `SDL_JoyAxisEvent` definition:
```c
typedef struct SDL_JoyAxisEvent
{
    Uint32 type;        /**< ::SDL_JOYAXISMOTION */
    Uint32 timestamp;
    SDL_JoystickID which; /**< The joystick instance id */
    Uint8 axis;         /**< The joystick axis index */
    Uint8 padding1;
    Uint8 padding2;
    Uint8 padding3;
    Sint16 value;       /**< The axis value (range: -32768 to 32767) */
    Uint16 padding4;
} SDL_JoyAxisEvent;
```

File: src/sdl2.nim [hunk 2]
```diff
@@ -116,15 +116,16 @@ type
...
+    which*: int32
     axis*: uint8
     pad1,pad2: uint8
     value*: cint
   JoyBallEventPtr* = ptr JoyBallEventObj
   JoyBallEventObj* = object
     kind*: EventType
     timestamp*: uint32
-    which*,ball*, pad1,pad2: uint8
```
Comment (yglukhov) [inline src/sdl2.nim]: ditto

File: src/sdl2.nim (hunk 1 above)
Comment (yglukhov) [inline src/sdl2.nim]: And `value` should be `int16` =)

File: src/sdl2.nim [hunk 3]
```diff
@@ -116,15 +116,16 @@ type
...
+    which*: int32
+    ball*, pad1,pad2: uint8
     xrel*,yrel*: int32
```
Comment (yglukhov) [inline src/sdl2.nim]: xrel and yrel should be `int16`

**Comment by ftsf** [comment]
Right you are!

**Comment by yglukhov** [comment]
Thank you!

**Pull request merged**